openfermioncirq/gates/four_qubit_gates.py

```diff
@@ -0,0 +1,139 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Gates that target four qubits."""
+
+from typing import Optional, Union
+
+import numpy
+
+import cirq
+
+
+class DoubleExcitationGate(cirq.EigenGate,
+                           cirq.CompositeGate,
+                           cirq.TextDiagrammable):
+    """Evolve under |0011><1100| + h.c. for some time."""
```

Comment:
I feel that the gate should actually be defined as evolving under the negative of this.